src/javascript/JContent/AdditionalAppsRoute/AdditionalAppsRoute.jsx

```diff
@@ -4,22 +4,34 @@ import {useNodeInfo} from '@jahia/data-helper';
 import {Route} from 'react-router';
 import React from 'react';
 import PropTypes from 'prop-types';
+import {Typography} from '@jahia/moonstone';
+import {useTranslation} from 'react-i18next';
+import styles from './AdditionAppsRoute.scss';
 
 export const AdditionalAppsRoute = ({match, target}) => {
     const site = useSelector(state => state.site);
+    const {t} = useTranslation('jcontent');
 
     const {routes: adminRoutes, allPermissions} = useAdminRouteTreeStructure(target);
-    const {node} = useNodeInfo({path: '/sites/' + site}, {getPermissions: allPermissions});
+    const {node} = useNodeInfo({path: '/sites/' + site}, {
+        getPermissions: allPermissions,
+        getSiteInstalledModules: true
+    });
 
     const filteredAdminRoutes = adminRoutes && adminRoutes
         .filter(route => route.requiredPermission === undefined || (node && (node[route.requiredPermission] !== false)))
-        .filter(route => route.isSelectable && route.render);
+        .filter(route => route.isSelectable && route.render)
+        .filter(route => route.requireModuleInstalledOnSite === undefined || (node && node.site.installedModulesWithAllDependencies.indexOf(route.requireModuleInstalledOnSite) !== -1));
 
     return (
         <>
             {filteredAdminRoutes && filteredAdminRoutes.map(r =>
                 <Route key={r.key} path={match.path + '/' + r.key} render={props => r.render(props)}/>
             )}
+            <Route key="nothingToDisplayRoute"
```

Comment:
This route will render in all cases, it should be wrapped in a <Switch> if we only want one to be used

Reply:
I'll check, but I thought that the routes returned were already part of a switch, maybe I didn't notice it because it rendered at the bottom. Anyways I'll check.

Reply:
You were right, it was rendering, there is no switch wrapping it. And it didn't show because iframe was covering it.